### DISCUSSION TITLE
Switch between two advertisements

### DISCUSSION THREAD

**Discussion by sivar2311** [h2zero/NimBLE-Arduino#371] [Q&A] [answered]
In my project I need two different advertisements to be sent out at different times.
One advertisement only contains manufacturer data. During the broadcast, it should not be possible to connect to the device.
The other advertisement is only sent out when the device allows connections and contains a service UUID.

Question: 
What is the easiest way to switch between two advertisements?
How can two NimBLEAdvertisement be used?

**Comment by h2zero** [comment, marked as answer]
The way I would do this is either create different functions to call for each advertisement type that sets the data. Or you could have different instances of `NimBLEAdvertisementData` that gets switched in.

**Comment by sivar2311** [comment]
Thanks for the tip with the `NimBLEAdvertisementData` - I will give that a try. 
(I wanted to avoid the repeated adding and deleting of the service uuid and the manufacturer data for the NimBLEAdvertisement).
